src/gui/torrentmodel.cpp

```diff
@@ -314,47 +298,53 @@ void TorrentModel::handleTorrentsUpdated()
 
 // Static functions
 
-QIcon getIconByState(BitTorrent::TorrentState state)
+static QString getIconId(BitTorrent::TorrentState state)
 {
     switch (state) {
     case BitTorrent::TorrentState::Downloading:
     case BitTorrent::TorrentState::ForcedDownloading:
     case BitTorrent::TorrentState::DownloadingMetadata:
-        return getDownloadingIcon();
+        return "downloading";
```

Comment:
I've gone with the approach of making the literals `static const QLatin1String` and I can then return a `const QLatin1String &` reference. That could open up other optimisations such as having `GuiIconProvider::getStatusIcon` take a `const QLatin1String &`, but then it would still need to be converted to `QString` internally (can't store `QLatin1String` in a hash as the `const char *` isn't copied). Need to think about this some more.